crates/ruff_linter/src/rules/flake8_simplify/rules/ast_bool_op.rs

```diff
@@ -326,34 +326,48 @@ pub(crate) fn duplicate_isinstance_call(checker: &mut Checker, expr: &Expr) {
             range: _,
         }) = &call
         else {
+            last_target_option = None;
             continue;
         };
         if args.len() != 2 {
+            last_target_option = None;
             continue;
         }
         if !keywords.is_empty() {
+            last_target_option = None;
             continue;
         }
         let Expr::Name(ast::ExprName { id: func_name, .. }) = func.as_ref() else {
+            last_target_option = None;
             continue;
         };
         if func_name != "isinstance" {
+            last_target_option = None;
             continue;
         }
         if !checker.semantic().is_builtin("isinstance") {
+            last_target_option = None;
```

Comment:
I'd love to avoid these but it's a bit tedious to do so...

One option: add a method like:

```rust
fn isinstance_target(call: &Expr, semantic: &SemanticModel) -> Option<&Expr> {
    // Verify that this is an `isinstance` call.
    let Expr::Call(ast::ExprCall {
        func,
        arguments:
            Arguments {
                args,
                keywords,
                range: _,
            },
        range: _,
    }) = &call
    else {
        return None;
    };
    if args.len() != 2 {
        return None;
    }
    if !keywords.is_empty() {
        return None;
    }
    let Expr::Name(ast::ExprName { id: func_name, .. }) = func.as_ref() else {
        return None;
    };
    if func_name != "isinstance" {
        return None;
    }
    if !semantic.is_builtin("isinstance") {
        return None;
    }

    // Collect the target (e.g., `obj` in `isinstance(obj, int)`).
    Some(&args[0])
}
```

Then this becomes:
```rust
let Some(target) = isinstance_target(&call, checker.semantic()) else {
    last_target_option = None;
    continue;
};
```

Reply:
Yes it's pretty repetitive, I will apply your change.

Reply:
Sounds good, will wait for that then merge. Thanks!